File: sdk/kotlin/src/main/kotlin/com/mulen/sdk/models/Balance.kt
package com.mulen.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Balance(
    @SerialName("shop_id")
    val shopId: Int,
    val currency: String,
    val balance: Double,
    val hold: Double,
    val available: Double
)

@Serializable
data class BalancesResponse(
    val success: Boolean,
    val data: BalancesData? = null
)

@Serializable
data class BalancesData(
    val balances: List<Balance>
)

File: sdk/flatter/mulen_pay_sdk/lib/src/models/balance.dart
class Balance {
  final int shopId;
  final String currency;
  final double balance;
  final double hold;
  final double available;

  Balance({
    required this.shopId,
    required this.currency,
    required this.balance,
    required this.hold,
    required this.available,
  });

  factory Balance.fromJson(Map<String, dynamic> json) {
    return Balance(
      shopId: json['shop_id'],
      currency: json['currency'],
      balance: json['balance'],
      hold: json['hold'],
      available: json['available'],
    );
  }
}
